iptstate.cc

```diff
@@ -1114,14 +1188,22 @@ int conntrack_hook(enum nf_conntrack_msg_type nf_type, struct nf_conntrack *ct,
     }
   }
 
-  if (flags->filter_dst) {
+  if (flags->filter_dst && !filters->has_dstnet) {
```

Comment:
I'm gonna go ahead and merge as-is, but I just realized we could make this code cleaner if instead of the:

```
if filster_dst && !destnet
...
if filterest && desntnet
...
```

We instead just did:

```
if (filter_dst) {
  if ((filter_inv && match_netmask(...)) || (!filter_inv && !match_netmask(...))) {
    ...
  }
}
```

and then inside match_netmask() we did:

```
if (!netmask) {
  return memcmp(...)
}
// rest of logic here
}
```

So all of the comparison of ip/nets was in one place and we weren't re-iffing it all the time.
    ..
  
  match_netmask